llvm/lib/Target/SPIRV/SPIRVInstructionSelector.cpp

```diff
@@ -2559,6 +2559,12 @@ bool SPIRVInstructionSelector::selectIntrinsic(Register ResVReg,
   } break;
   case Intrinsic::spv_saturate:
     return selectSaturate(ResVReg, ResType, I);
+  case Intrinsic::spv_fclamp:
+    return selectExtInst(ResVReg, ResType, I, CL::fclamp, GL::FClamp);
```

Comment:
I don't beleive fclamp is correct for HLSL. I beleive HLSL uses NClamp since max and min are NMin and NMax.  
We had to fix a bug like this in the past.
https://github.com/llvm/llvm-project/issues/87072

Either we make a spv_nclamp and use that in CGBuiltins.cpp or we just change this line to:
```suggestion
    return selectExtInst(ResVReg, ResType, I, CL::nclamp, GL::NClamp);
```

Reply:
looks like `CL::nclamp` doesn't exist, but `CL::fclamp` actually has the same behaviour as `GL::NClamp`: in fmin/fmax, if one of the args is NaN, it returns the other argument.

Reply:
Anyways, it's now doing the right thing and tests verify it.